include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -800,6 +880,74 @@ namespace VectorTools
     }
 
 
+
+    /**
+     * Generic implementation for the project() function on distributed triangulations.
+     */
+    template <int dim, int spacedim, typename VectorType, int components, int fe_degree>
+    void do_project_distributed (const Mapping<dim, spacedim>                              &mapping,
+                                 const DoFHandler<dim, spacedim>                           &dof,
+                                 const ConstraintMatrix                                    &constraints,
+                                 const Quadrature<dim>                                     &quadrature,
+                                 const Function<spacedim, typename VectorType::value_type> &function,
+                                 VectorType                                                &vec_result,
```

Comment:
please document whether `vec_result` should be fully distributed or ghosted or something else.